addons/mrp/models/mrp_production.py

```diff
@@ -1395,13 +1395,15 @@ def _generate_backorder_productions(self, close_mo=True):
                 })
             else:
                 for move in production.move_raw_ids | production.move_finished_ids:
-                    new_move = self.env['stock.move'].browse(move._split(move.product_uom_qty - move.unit_factor * production.qty_producing))
-                    if move.raw_material_production_id:
-                        new_move.raw_material_production_id = backorder_mo.id
-                    else:
-                        new_move.production_id = backorder_mo.id
-                    (move | new_move)._do_unreserve()
-                    (move | new_move)._action_assign()
+                    if move.bom_line_id or not move.additional or move in production.move_finished_ids:
```

Comment:
You could remove move in production.move_finished_ids and do for move in production.move_raw_ids.
We don't want to split finished move anymore? This PR should only contains the
```
if not move.additional
```